x-pack/plugins/transform/public/app/hooks/use_get_data_views_title_id_map.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import type { IHttpFetchError } from '@kbn/core-http-browser';
+
+import { TRANSFORM_REACT_QUERY_KEYS } from '../../../common/constants';
+
+import { useAppDependencies } from '../app_dependencies';
+
+type DataViewListTitleIdMap = Record<string, string>;
+
+export const useGetDataViewsTitleIdMap = () => {
+  const { data } = useAppDependencies();
+
+  return useQuery<DataViewListTitleIdMap, IHttpFetchError>(
+    [TRANSFORM_REACT_QUERY_KEYS.GET_DATA_VIEW_IDS_WITH_TITLE],
+    async () => {
+      // Since we let useQuery take care of caching,
+      // clear the cache to ensure we get the latest data view list.
+      data.dataViews.clearCache();
```

Comment:
❓ can it influence the performance of other plugins that rely on the data views caching provided by the shared service?

Reply:
Looked into this again and turns out `dataViews.getIdsWithTitle(true)` is how you can refresh just this specific call so I was able to get rid of the whole `data.dataViews.clearCache();` again in [928e66c](https://github.com/elastic/kibana/pull/181892/commits/928e66c384978bba53b25a8c4a8f9c9a6a3476d3).